blueflood-core/src/main/java/com/rackspacecloud/blueflood/utils/Metrics.java

```diff
@@ -159,4 +159,12 @@ public static Histogram histogram(Class kls, String... names) {
     public static Counter counter(Class kls, String... names) {
         return getRegistry().counter(MetricRegistry.name(kls, names));
     }
+
+    public static void registerGauge(Class kls, Gauge gauge, String ...names) {
+        // MetricRegistry does not offer getOrAdd wrapper for Gauge
```

Comment:
Isn't that because it would be an error to register the same gauge more than once? With the code you've added here, if someone were to create a second gauge with the same name somewhere else, it would be silently dropped.

Reply:
MetricRegistry#getOrAdd does the same. It ignores if its the same Metric to be registered. getOrAdd has protected access specifier. So I had to write similar wrapper for Gauge.
Gauge name includes the class name to make it namespace specific. So duplicate chances are less. If it still ends up duplicate, it gets ignored. Similar to other metric types.

Reply:
aah I see the concern for Gauge. If another Gauge with different getValue implementation is added, it gets silently ignored. But the chances of collision are less due to Class name part of the Gauge name. Do you suggest raise Exception if it already exists?

Reply:
You're right that the chances of collision are low. It seems that the MetricRegistry already prevents duplicate registrations:
```
            if (existing == null) {
                onMetricAdded(name, metric);
            } else {
                throw new IllegalArgumentException("A metric named " + name + " already exists");
            }
```
so I'd just let it do its thing and not over-code here.

Reply:
I'm pretty sure the reason there's not a comparable "getOrAdd" function for gauges is that they're fundamentally different than other metric types. They can be looked up and updated at any time to send new data, but a gauge is always created once and passively monitors a thing. There's no reason to ever "get" a gauge after you've registered it.

Reply:
Unfortunately we can't use this as is. We have 3 instances of the ElasticsearchRestHelper. So 3 constructor calls->3 times metrics registration attempts leading to exceptions. Unless we make ElasticsearchRestHelper singleton, we can't use MetricRegistry#register as is. I have to put try/catch around every gauge registration otherwise.

Reply:
Reminder: we decided in Zoom to go ahead and change this to a static singleton, and we'll try to get a feel for any performance issues in staging.